x-pack/plugins/integration_assistant/server/routes/categorization_routes.ts

```diff
@@ -36,64 +39,53 @@ export function registerCategorizationRoutes(
         },
       },
-      async (context, req, res) => {
-        const { packageName, dataStreamName, rawSamples, currentPipeline } =
-          req.body as CategorizationApiRequest;
-
+      async (context, req, res): Promise<IKibanaResponse<CategorizationResponse>> => {
+        const { packageName, datastreamName, rawSamples, currentPipeline } = req.body;
         const services = await context.resolve(['core']);
         const { client } = services.core.elasticsearch;
         const { getStartServices, logger } = await context.integrationAssistant;
         const [, { actions: actionsPlugin }] = await getStartServices();
-        const actionsClient = await actionsPlugin.getActionsClientWithRequest(req);
-        const connector = req.body.connectorId
-          ? await actionsClient.get({ id: req.body.connectorId })
-          : (await actionsClient.getAll()).filter(
-              (connectorItem) => connectorItem.actionTypeId === '.bedrock'
-            )[0];
 
-        const abortSignal = getRequestAbortedSignal(req.events.aborted$);
-        const isOpenAI = connector.actionTypeId === '.gen-ai';
-        const llmClass = isOpenAI ? ActionsClientChatOpenAI : ActionsClientSimpleChatModel;
+        try {
+          const actionsClient = await actionsPlugin.getActionsClientWithRequest(req);
+          const connector = req.body.connectorId
+            ? await actionsClient.get({ id: req.body.connectorId })
+            : (await actionsClient.getAll()).filter(
+                (connectorItem) => connectorItem.actionTypeId === '.bedrock'
+              )[0];
+
+          const abortSignal = getRequestAbortedSignal(req.events.aborted$);
+          const isOpenAI = connector.actionTypeId === '.gen-ai';
+          const llmClass = isOpenAI ? ActionsClientChatOpenAI : ActionsClientSimpleChatModel;
 
-        const model = new llmClass({
-          actions: actionsPlugin,
-          connectorId: connector.id,
-          request: req,
-          logger,
-          llmType: isOpenAI ? 'openai' : 'bedrock',
-          model: req.body.model || connector.config?.defaultModel,
-          temperature: 0.05,
-          maxTokens: 4096,
-          signal: abortSignal,
-          streaming: false,
-        });
+          const model = new llmClass({
+            actions: actionsPlugin,
+            connectorId: connector.id,
+            request: req,
+            logger,
+            llmType: isOpenAI ? 'openai' : 'bedrock',
+            model: connector.config?.defaultModel,
+            temperature: 0.05,
+            maxTokens: 4096,
+            signal: abortSignal,
+            streaming: false,
+          });
 
-        const graph = await getCategorizationGraph(client, model);
-        let results = { results: { docs: {}, pipeline: {} } };
-        try {
-          results = (await graph.invoke({
+          const graph = await getCategorizationGraph(client, model);
+          const results: CategorizationResponse = await graph.invoke({
```

Comment:
To be 100% sure about the return type you could validate the response like

```ts
const results = await graph.invoke({
  packageName,
  dataStreamName,
  datastreamName,
  rawSamples,
  currentPipeline,
}));

return res.ok({ body: CategorizationApiResponse.parse(results) });
```

Reply:
Changed in https://github.com/elastic/kibana/pull/186085/commits/ceb8d3e560a914767431617127638289d6e0a3ad